load_test.py

```diff
@@ -14,22 +14,43 @@
 import utils
 
 
-def run_main_process(concurrency, duration, dataset, dataset_q, stop_q):
+def run_main_process(rps, duration, dataset, request_q, stop_q):
     """Run the main process."""
     logging.info("Test from main process")
 
-    # Initialize the dataset_queue with 4*concurrency requests
+    start_time = time.time()
+    end_time = start_time + duration
+    if rps is not None:
+        main_loop_rps_mode(dataset, request_q, rps, start_time, end_time)
+    else:
+        main_loop_concurrency_mode(dataset, request_q, start_time, end_time)
+
+    logging.info("Timer ended, stopping processes")
+
+    # Signal users to stop sending requests
+    stop_q.put(None)
+
+    # Empty the dataset queue
+    while not request_q.empty():
+        logging.debug("Removing element from request_q")
+        request_q.get()
+
+    return
+
+def main_loop_concurrency_mode(dataset, request_q, start_time, end_time):
+    """Let all users send requests repeatedly until end_time"""
+    logging.info("Test from main process")
```

Comment:
Do we still need this logging statement here?

Reply:
No, I'll remove this thanks!